src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -58,6 +70,64 @@ export const DiscoverTopNav = ({
     [history, resetSavedSearch, savedSearch.id]
   );
 
+  useEffect(() => {
+    return () => {
+      // Make sure to close the editors when unmounting
+      if (closeFieldEditor.current) {
+        closeFieldEditor.current();
+      }
+      if (closeDataViewEditor.current) {
+        closeDataViewEditor.current();
+      }
+    };
+  }, []);
+
+  const editField = useMemo(
+    () =>
+      canEditDataViewField
+        ? async (fieldName?: string, uiAction: 'edit' | 'add' = 'edit') => {
```

Comment:
Can we make this a bit more functional and move this out into it's own function with `try..catch` and an `abortController`? It'll make it easier to read

Reply:
This code existed in Discover, I just moved it in another file. Exactly same code existed in Lens. I am thinking of cleaning this up a bit but not on this PR